sapi/cgi/tests/012.phpt

```diff
@@ -0,0 +1,114 @@
+--TEST--
+multiple files syntax check
+--SKIPIF--
+<?php include "skipif.inc"; ?>
+--INI--
+display_errors=stdout
+--FILE--
+<?php
+include "include.inc";
+
+function run_and_output($cmd) {
+    if (!defined("PHP_WINDOWS_VERSION_MAJOR")) {
+        $cmd .= " 2>/dev/null";
+    }
+    exec($cmd, $output, $exit_code);
+    print_r($output);
+    var_dump($exit_code == 0);
+}
+
+$php = get_cgi_path();
+reset_env_vars();
+
+$filename_good = __DIR__."/012_good.test.php";
+$filename_good_escaped = escapeshellarg($filename_good);
+$filename_bad = __DIR__."/012_bad.test.php";
+$filename_bad_escaped = escapeshellarg($filename_bad);
+
+$code = '
+<?php
+
+echo "hi";
+';
+
+file_put_contents($filename_good, $code);
+
+run_and_output("$php -n -l $filename_good_escaped $filename_good_escaped");
+run_and_output("$php -n -l $filename_good_escaped some.unknown $filename_good_escaped");
+
+$code = '
+<?php
+
+class test
+    private $var;
+}
+
+?>
+';
+
+file_put_contents($filename_bad, $code);
+
+run_and_output("$php -n -l $filename_good_escaped $filename_bad_escaped $filename_good_escaped");
+run_and_output("$php -n -l $filename_bad_escaped $filename_bad_escaped");
+run_and_output("$php -n -l $filename_bad_escaped some.unknown $filename_bad_escaped");
+run_and_output("$php -n -l $filename_bad_escaped $filename_bad_escaped some.unknown");
+
+echo "Done\n";
+?>
+--CLEAN--
+<?php
+@unlink($filename_good);
+@unlink($filename_bad);
+?>
+--EXPECTF--
+Array
+(
+    [0] => No syntax errors detected in %s012_good.test.php
+    [1] => No syntax errors detected in %s012_good.test.php
+)
+bool(true)
+Array
+(
+    [0] => No syntax errors detected in %s012_good.test.php
+    [1] => No input file specified.
```

Comment:
This differs in behavior with CLI, in that it aborts when a file is not found. Is this a limitation?

Reply:
Yeah we'd need to change error handling in lines 2451-2488: if a file cannot be opened or accessed CGI aborts. But I'm not sure if we should special case it because it will make things more complicated and possibly slow down regular use of CGI. Let me know what you think.

Reply:
I doubt linting is often used with php-cgi, so this should be ok.